src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -380,25 +380,17 @@ void mp::QemuVirtualMachine::shutdown(bool force)
             mpl::log(mpl::Level::debug, vm_name, "No process to kill");
         }
 
-        if (mp::backend::instance_image_has_snapshot(desc.image.image_path, suspend_tag))
-        {
-            if (state != State::suspended)
-            {
-                mpl::log(
-                    mpl::Level::warning,
-                    vm_name,
-                    fmt::format("{} has the image suspension snapshot, but it is not in the suspended state", vm_name));
-            }
+        const auto has_suspend_snapshot = mp::backend::instance_image_has_snapshot(desc.image.image_path, suspend_tag);
+        if (has_suspend_snapshot != (state == State::suspended)) // clang-format off
+            mpl::log(mpl::Level::warning, vm_name, fmt::format("Image has {} suspension snapshot, but the state is {}",
+                                                               has_suspend_snapshot ? "a" : "no",
+                                                               static_cast<short>(state))); // clang-format on
 
```

Comment:
The only problem of this is the integer value returns to us via `static_cast<short>(state)`, however, enum -> string map or a switch statement may not worth it.

Reply:
Right, this is what my comment above about formatters is about. Ideally, we'd write a Formatter for VM::State... see above.